examples/two-columns-layout/src/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { BaseItem, OnActiveParams } from '@algolia/autocomplete-core';
+import { dequal } from 'dequal/lite';
+
+type smartPreviewParams<TItem extends BaseItem> = OnActiveParams<TItem> & {
+  contextData: {
+    [key: string]: unknown;
+  };
+};
+
+export const updateActiveItem = (activeItemId: number) => {
+  const currentActiveEl = document.querySelector('[data-active=true]');
+  currentActiveEl?.removeAttribute('data-active');
+
+  const selectedEl = document.querySelector(
+    `#autocomplete-0-item-${activeItemId}`
+  );
+  selectedEl?.setAttribute('data-active', 'true');
+};
```

Comment:
Why is this needed? We shouldn't directly update the DOM and instead rely on DOM updates derived from state updates.

Maybe the `refresh` call happens too soon in your function below?

Reply:
The item attribute `aria-selected="true"` is already used by Autocomplete but it appears only when you are hovering an item.
With the smart preview, I need to keep the current previewed item highlighted, so I used a new attribute `data-active`, but I know it's not optimal indeed since I rely on the DOM.